src/test/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpointIT.java

```diff
@@ -1203,4 +1223,161 @@ public void shouldGetCaseTypesForReadAccess() throws Exception {
                 hasProperty("id", equalTo("PersonAddress"))))
         );
     }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_case_event_history.sql"})
+    public void shouldGetCaseHistoryForEvent() throws Exception {
+
+        // Check that we have the expected test data set size
+        List<CaseDetails> resultList = template.query("SELECT * FROM case_data", this::mapCaseData);
+        assertEquals("Incorrect data initiation", 1, resultList.size());
+
+        List<AuditEvent> eventList = template.query("SELECT * FROM case_event", this::mapAuditEvent);
+        assertEquals("Incorrect data initiation", 3, eventList.size());
+
+        MvcResult result = mockMvc.perform(get(GET_CASE_HISTORY_FOR_EVENT + eventList.get(1).getId())
+            .contentType(JSON_CONTENT_TYPE)
+            .header(AUTHORIZATION, "Bearer user1"))
+            .andExpect(status().is(200))
+            .andReturn();
+
+        final CaseHistoryView caseHistoryView = mapper.readValue(result.getResponse().getContentAsString(), CaseHistoryView.class);
+        assertNotNull("Case View is null", caseHistoryView);
+        assertEquals("Unexpected Case ID", Long.valueOf(1504259907353529L), Long.valueOf(caseHistoryView.getCaseId()));
+
+        final CaseViewType caseViewType = caseHistoryView.getCaseType();
+        assertNotNull("Case View Type is null", caseViewType);
+        assertEquals("Unexpected Case Type Id", "TestAddressBookCase", caseViewType.getId());
```

Comment:
not a big issue you can ignore this but I think we should favour assertThat because it gives better failure messages